### PR TITLE
bsp: Allow to change size of the draw buffer

### PR DESCRIPTION
**Pull request by espzav**
After discussion, there should be way, how to change the draw buffer size in bsp. This is the solution.

### CODE REVIEW

**Comment by espzav** [comment]
@tore-espressif PTAL

File: esp-box/include/bsp/esp-box.h [hunk 1]
```diff
@@ -274,9 +274,12 @@ esp_err_t bsp_spiffs_unmount(void);
 #define BSP_LCD_PIXEL_CLOCK_HZ     (40 * 1000 * 1000)
 #define BSP_LCD_SPI_NUM            (SPI3_HOST)
 
-#define BSP_LCD_DRAW_BUFF_SIZE     (BSP_LCD_H_RES * 10)
 #define BSP_LCD_DRAW_BUFF_DOUBLE   (0)
 
+#ifndef BSP_LCD_DRAW_BUFF_SIZE
```
Comment (igrr) [inline esp-box/include/bsp/esp-box.h:279]: Just out of curiosity, why is it not a value which is set at run time? We aren't allocating this buffer statically anyway, right? (Or, if making it a compile time setting, why not via a Kconfig option?)

**Comment by espzav** [comment]
> Hello @espzav , so how is the user supposed to configure `BSP_LCD_DRAW_BUFF_SIZE`?

When you define the `BSP_LCD_DRAW_BUFF_SIZE` before the `esp-box.h` hedaer file.

**Comment by espzav** [comment]
> > > Hello @espzav , so how is the user supposed to configure `BSP_LCD_DRAW_BUFF_SIZE`?
> > 
> > 
> > @tore-espressif When you define the `BSP_LCD_DRAW_BUFF_SIZE` before the `esp-box.h` hedaer file.
> 
> Ah, but this way, the definition will never make its way to `esp_box.c`, right? :D

 @tore-espressif yes, you are right. Then, you can define into Cmake :-) I absolutely agree, that this is not good solution. I am sorry. It seems, that we must change API and add it into runtime params :-(

File: esp-box/include/bsp/esp-box.h (hunk 1 above)
Comment (tore-espressif) [reply esp-box/include/bsp/esp-box.h:279]: I am in favor of the first solution -> runtime settings.

The initial API was drafted with an idea of as simple as possible functions, often with no parameters, so the user could just call a few functions to initialize his development board. The requirement has changed now, as we want to ship BSP with various GUI projects. Let's now introduce a breaking change, that will be open for extensions and closed for modifications :)

Comment (espzav) [reply esp-box/include/bsp/esp-box.h:279]: @tore-espressif Ok, I am working on it. Thank you

**Comment by espzav** [comment]
@igrr and @tore-espressif I changed it to runtime API. I made it only for one board, please, could you check it and I will copy it to all BSPs after that.

**Comment by tore-espressif** [comment]
> @igrr and @tore-espressif I changed it to runtime API. I made it only for one board, please, could you check it and I will copy it to all BSPs after that.

So the LVGL users that call `bsp_display_start` will not have an option to change the draw buffer size, right?

**Comment by espzav** [comment]
> > @igrr and @tore-espressif I changed it to runtime API. I made it only for one board, please, could you check it and I will copy it to all BSPs after that.
> 
> So the LVGL users that call `bsp_display_start` will not have an option to change the draw buffer size, right?

Yes, that's right. The important is allow to change the transfer size in `bsp_display_new`. Do you think, that we should add it into  `bsp_display_start`?

File: esp_wrover_kit/include/bsp/display.h [hunk 2]
```diff
@@ -55,13 +63,23 @@ extern "C" {
...
+esp_err_t bsp_display_new(bsp_display_config_t *config, esp_lcd_panel_handle_t *ret_panel, esp_lcd_panel_io_handle_t *ret_io);
+
+/**
+ * @brief BSP display default configuration
+ *
  */
-esp_err_t bsp_display_new(esp_lcd_panel_handle_t *ret_panel, esp_lcd_panel_io_handle_t *ret_io);
+#define BSP_DISPLAY_CONFIG()           \
+    {                                       \
+        .max_transfer_sz = BSP_LCD_DRAW_BUFF_SIZE*sizeof(uint16_t), \
```
Comment (tore-espressif) [inline esp_wrover_kit/include/bsp/display.h:81]: This definition relies on `BSP_LCD_DRAW_BUFF_SIZE`, which is defined in different header file: `esp_wrover_kit.h`

**Review by tore-espressif — CHANGES_REQUESTED** [review]
Only one comment, otherwise LGTM. Let's wait for @igrr

File: esp_wrover_kit/include/bsp/display.h (hunk 2 above)
Comment (espzav) [reply esp_wrover_kit/include/bsp/display.h:82]: Right, I included header `esp-bsp.h`.

**Comment by espzav** [comment]
@tore-espressif I reworked it. Now it is changed for all boards. PTAL

**Review by tore-espressif — APPROVED** [review]
LGTM!

**Pull request merged**